keg_storage/backends/azure.py

```diff
@@ -119,20 +133,26 @@ class AzureReader(AzureFile):
     """
     The Azure reader uses byte ranged API calls to fill a local buffer to avoid lots of API overhead
     for small read sizes.
+
     """
 
     def __init__(
         self,
         mode: base.FileMode,
         blob_client: BlobClient,
         chunk_size=DEFAULT_CHUNK_SIZE,
+        validate_content=False,
     ):
+        """
+        :param validate_content: If set to true, the downloaded blob will be validated using the
+            stored Content-MD5 property.
+        """
         super().__init__(
             mode=mode,
             blob_client=blob_client,
             chunk_size=chunk_size,
         )
-        self.stream = self.client.download_blob()
+        self.stream = self.client.download_blob(validate_content=validate_content)
```

Comment:
I tested this with a blob that had a deliberately invalid MD5 digest, and it worked fine. So something isn't quite right here.